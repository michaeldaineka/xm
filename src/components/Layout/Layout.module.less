@import 'styles/app';

.layout {
  display: flex;
  .content {
    display: flex;
    flex-direction: column;
    margin-top: 85px;
    width: 100%;
    :global(.error) {
      margin: -30px;
      width: calc(100% + 60px);
    }
  }
}
