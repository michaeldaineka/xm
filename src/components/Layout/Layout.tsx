import React, { Suspense } from 'react';
import { Switch, Route } from 'react-router-dom';
import s from './Layout.module.less';
import Header from 'components/Header';
import layoutSchema from './schema/layoutSchema';

const Layout = (): JSX.Element => {
  return (
    <div className={s.layout}>
      <Header />
      <main className={s.content}>
        <Suspense fallback={<p>loading...</p>}>
          <Switch>
            {layoutSchema.map((item) => (
              <Route {...item} key={Math.floor(Math.random() * 3)} />
            ))}
          </Switch>
        </Suspense>
      </main>
    </div>
  );
};

export default Layout;
