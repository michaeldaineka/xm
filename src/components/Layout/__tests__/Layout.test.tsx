import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../Layout';
import {BrowserRouter} from "react-router-dom";

test('Layout component renders', () => {
    const component = renderer.create(
        <BrowserRouter>
            <Layout/>
        </BrowserRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
