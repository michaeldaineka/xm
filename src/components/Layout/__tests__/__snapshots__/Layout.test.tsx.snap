// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component renders 1`] = `
<div>
  <header>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col d-flex ant-col-sm-24"
        style={Object {}}
      >
        <img
          alt="XM Logo"
          height="39"
          src="logo.svg"
          width="147.88px"
        />
        <small
          className={[Function]}
        >
          task
        </small>
      </div>
    </div>
  </header>
  <main>
    <p>
      loading...
    </p>
  </main>
</div>
`;

exports[`Layout component renders 2`] = `
<div>
  <header>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col d-flex ant-col-sm-24"
        style={Object {}}
      >
        <img
          alt="XM Logo"
          height="39"
          src="logo.svg"
          width="147.88px"
        />
        <small
          className={[Function]}
        >
          task
        </small>
      </div>
    </div>
  </header>
  <main>
    <p>
      loading...
    </p>
  </main>
</div>
`;

exports[`Layout component renders 3`] = `
<div>
  <header>
    <div
      className="ant-row"
      style={Object {}}
    >
      <div
        className="ant-col d-flex ant-col-sm-24"
        style={Object {}}
      >
        <img
          alt="XM Logo"
          height="39"
          src="logo.svg"
          width="147.88px"
        />
        <small
          className={[Function]}
        >
          task
        </small>
      </div>
    </div>
  </header>
  <main>
    <p>
      loading...
    </p>
  </main>
</div>
`;
